src/frontend/js/widgets/Dashboard/components/DashboardItem/Contract/index.tsx

```diff
@@ -35,13 +43,14 @@ export const DashboardItemContract = ({
             <span>{contract_definition.title}</span>
           </div>
           <div className="dashboard-contract__footer">
-            <span>
+            <span className="dashboard-contract__footer__status">
               <ContractStatus contract={contract} />
             </span>
             <div>
-              {contract && contract.student_signed_on ? (
-                <DownloadContractButton contract={contract} />
-              ) : (
+              {contractState === ContractState.SIGNED && (
+                <DownloadContractButton contract={contract!} className="dashboard-item__button" />
+              )}
+              {contractState === ContractState.UNSIGNED && (
```

Comment:
This is already managed by the `SignContractButton` 🤔

Reply:
nice catch, i've fail my rebase and add back something that should not be there.